src/test/java/org/opentripplanner/middleware/triptracker/TravelerLocatorTest.java

```diff
@@ -0,0 +1,38 @@
+package org.opentripplanner.middleware.triptracker;
+
+import org.junit.jupiter.api.Test;
+import org.opentripplanner.middleware.otp.response.Leg;
+import org.opentripplanner.middleware.otp.response.Place;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class TravelerLocatorTest {
+    @Test
+    void testStopsUntilEndOfLeg() {
+        Leg leg = new Leg();
+        leg.to = createPlace("FinalStop");
+        leg.intermediateStops = List.of(
+            createPlace("Stop0"),
+            createPlace("Stop1"),
+            createPlace("Stop2"),
+            createPlace("Stop3"),
+            createPlace("Stop4"),
+            createPlace("Stop5"),
+            createPlace("Stop6")
+        );
+
+        for (int i = 0; i < 7; i++) {
```

Comment:
Could use `leg.intermediateStops.size()` instead of 7.

Reply:
Thanks - updated in https://github.com/ibi-group/otp-middleware/pull/238/commits/81233642c82bdcb7832951412d1178d2f666d43f.